playground/src/App.tsx

```diff
@@ -16,6 +17,75 @@ async function getExample(example: string) {
     return [(await inputs).text(), (await masm).text()];
   }
 
+/**
+* We check whether the input is a valid JSON.
+* Or if it is empty.
+*/ 
+
+function tryParseJSONObject(jsonString: string) {
+  if (jsonString === "") {
+    return true;
+  }
+  try {
+      const o = JSON.parse(jsonString);
+
+      // Handle non-exception-throwing cases:
+      // Neither JSON.parse(false) or JSON.parse(1234) throw errors, hence the type-checking,
+      // but... JSON.parse(null) returns null, and typeof null === "object", 
+      // so we must check for that, too. Thankfully, null is falsey, so this suffices:
+      if (o && typeof o === "object") {
+          return true;
+      }
+  }
+  catch (e) {
+    toast.error('Miden VM Inputs need to be a valid JSON object.');
+  }
+  
+  return false;
+
+}
+
+/**
+* We allow an empty input.
+* If it is not empty, we check whether the input has 
+* either stack_init or advice_tape.
+* Then we check whether the values are numbers.
+*/ 
+
+function checkStackInitAndAdviceTape(jsonString: string) { 
+
+  if (jsonString === "") { 
+    return true 
+  }
+
+  if (!jsonString.includes("stack_init") && !jsonString.includes("advice_tape")) {
+    toast.error('Miden VM Inputs can be empty or we need either a stack_init or an advice_tape.');
+    return false
+  }
+
+  const o = JSON.parse(jsonString);
```

Comment:
There's some duplication between these. Is there a reason we can't combine them into a single `checkInputs` function?

Reply:
jup that works as well